tactics/tactics.ml

```diff
@@ -2993,16 +2993,29 @@ let quantify lconstr =
    solve, ui are a mix of inferred args and yi. The overall effect
    is to remove from H as much quantification as possible given
    lbind. *)
+
+(* let subst a b et = Termops.replace_term sigma a b et in *)
+let rec map_rel_context_lift f env i (ctxt:EConstr.rel_context):EConstr.rel_context =
+  match ctxt with
+  | [] -> ctxt
+  | decl::ctxt' -> f i decl :: map_rel_context_lift f env (i+1) ctxt'
```

Comment:
this is my way of doing a map that takes the depth into account lifting things in a rel_context. Is there a cleaner way?

Reply:
Not that I know of in the current API for contexts, but you could use the list API instead.

Reply:
Strange, I would have bet that performing rel-aware substitutions in a rel_context was something usual (but maybe that is more something we want to avoid?).
I could indeed use a fold.